src/components/Todo.js

```diff
@@ -1,6 +1,10 @@
-import React from "react";
+import React, { useState } from "react";
+import DetailsForm from './DetailsForm'
 const Todo = (props) => {
-  const { id, text, todo, todos, setTodos, details } = props;
+  console.log("props are",props)
+
+  const {  text, todo: {id, ...todo}, todos, setTodos, details } = props; //id,
```

Comment:
id was not present in props at the first level, so it will always be `undefined`, make sure the destructed data is present at the first level in props. if the data is present as a nested value then destructure as we are destructuring todo.id above.  @raiab